packages/components/src/Datalist/Datalist.component.js

```diff
@@ -0,0 +1,341 @@
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Select an item in suggestions list
+	 * @param event The select event
+	 * @param itemIndex The item index in suggestions list
+	 */
+	onSelect(event, { sectionIndex, itemIndex }) {
+		let newValue = this.state.groups[itemIndex];
+		if (this.props.multiSection) {
+			newValue = this.state.groups[sectionIndex].suggestions[itemIndex];
+		}
+		this.updateValue(event, newValue, true);
+	}
+
+	updateSelectedIndexes(value) {
+		if (this.props.multiSection) {
+			this.props.titleMap.forEach((group, sectionIndex) => {
+				const focusedIndex = group.suggestions.findIndex(item => item.name === value);
+				if (focusedIndex > -1) {
+					this.setState({
+						focusedItemIndex: focusedIndex,
+						focusedSectionIndex: sectionIndex,
+					});
+				}
+			});
+		} else {
+			const index = this.props.titleMap.findIndex(item => item.name === value);
+			this.setState({
+				focusedItemIndex: index,
+			});
+		}
+	}
+
+	/**
+	 * Set a value.
+	 * This new value can be persisted, or not. If not, it enables the ESC key to reset the value
+	 * @param event The change event
+	 * @param value The new value
+	 * @param persist Value will be persisted if true
+	 */
+	updateValue(event, value, persist) {
+		const previousValue = persist ? value : this.state.previousValue;
+
+		this.setState({
+			// setting the filtered value so it needs to be actual value
+			value: typeof value === 'object' ? value.title : value,
+			previousValue: typeof previousValue === 'object' ? previousValue.title : previousValue,
+		});
+		if (persist) {
+			let enumValue = this.props.titleMap.find(item => item.name === value);
+			if (this.props.multiSection) {
+				this.props.titleMap.forEach(group => {
+					const itemObj = group.suggestions.find(item => item.name === value.title);
+					if (itemObj) {
+						enumValue = itemObj;
+					}
+				});
+			}
+			const payload = { value: get(enumValue, 'value', value) };
+			this.props.onChange(event, payload);
+			this.props.onFinish(event, payload);
+		}
+	}
+
+	/**
+	 * Set back the value to the last validated value.
+	 * This remove the suggestions.
+	 */
+	resetValue() {
+		this.resetSuggestions();
+		this.setState({
+			value: this.state.previousValue,
+		});
+	}
+
+	/**
+	 * Building multiSection items or single section items
+	 * return the items list
+	 */
+	buildGroupItems() {
+		if (this.props.multiSection) {
+			return this.props.titleMap.map(group => ({
+				title: group.title,
+				suggestions: group.suggestions.map(item => ({ title: item.name })),
+			}));
+		}
+		return this.props.titleMap.map(item => item.name);
+	}
+
+	/**
+	 * Filter suggestions.
+	 * This sets at least an empty array, which means the suggestion box will always display
+	 * If the array is empty, the suggestion box will display a "No result" message
+	 * @param value The value to base suggestions on
+	 */
+	updateGroups(value, filterOn) {
```

Comment:
`filterOn` is useless, test if you have a value for that. And depending on the case, if you want to filter the result, pass the value, if not, pass nothing

Please rename into `updateSuggestions`. Here you update the suggestions, with or without groups. You don't update groups.